akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerMTSpec.scala

```diff
@@ -53,39 +53,43 @@ class CircuitBreakerMTSpec extends AkkaSpec {
     }
 
     "allow many calls while in closed state with no errors" in {
-      val futures = testCallsWithBreaker()
+      val futures = testCallsWithBreaker(newBreaker)
       val result = Await.result(Future.sequence(futures), 5.second.dilated)
       result.size should ===(numberOfTestCalls)
       result.toSet should ===(Set("succeed"))
     }
 
     "transition to open state upon reaching failure limit and fail-fast" in {
-      openBreaker()
-      val futures = testCallsWithBreaker()
+      val breaker = newBreaker
+      openBreaker(breaker)
+      val futures = testCallsWithBreaker(breaker)
       val result = Await.result(Future.sequence(futures), 5.second.dilated)
       result.size should ===(numberOfTestCalls)
       result.toSet should ===(Set("CBO"))
     }
 
     "allow a single call through in half-open state" in {
+      val breaker = newBreaker
       val halfOpenLatch = new TestLatch(1)
       breaker.onHalfOpen(halfOpenLatch.countDown())
 
-      openBreaker()
+      openBreaker(breaker)
 
       // breaker should become half-open after a while
       Await.ready(halfOpenLatch, resetTimeout + 1.seconds.dilated)
 
-      val futures = testCallsWithBreaker()
+      val futures = testCallsWithBreaker(breaker)
```

Comment:
here we want to verify we are in half-open. We want to see one call passing and others failing. 

I found the logic inside `testCallsWithBreaker` quite obscure. We call it 100 times, but the first 5 calls gets delayed with the latch. The others run faster because the latch reaches 0, which also unlocks the first 5.